src/cli/pm_pkg_command.zig

```diff
@@ -0,0 +1,784 @@
+                        Global.exit(1);
+                    }
+                }
+                if (err != error.NotFound) return err;
+            }
+        }
+
+        if (results.count() == 0) {
+            Output.println("{{}}", .{});
+        } else if (results.count() == 1) {
+            const value = results.values()[0];
+            Output.println("{s}", .{value});
+        } else {
+            Output.println("{{", .{});
+            for (results.keys(), results.values(), 0..) |key, value, i| {
+                const comma = if (i == results.count() - 1) "" else ",";
+                Output.println("  \"{s}\": {s}{s}", .{ key, value, comma });
+            }
+            Output.println("}}", .{});
+        }
+    }
+
+    fn execSet(ctx: Command.Context, pm: *PackageManager, args: []const string, cwd: []const u8) !void {
+        if (args.len == 0) {
+            Output.errGeneric("<blue>bun pm pkg set<r> expects a key=value pair of args", .{});
+            Global.exit(1);
+        }
+
+        const parse_json = pm.options.json_output;
+
+        const path = try findPackageJson(ctx.allocator, cwd);
+        defer ctx.allocator.free(path);
+
+        const pkg = try loadPackageJson(ctx, ctx.allocator, path);
+        defer ctx.allocator.free(pkg.contents);
+
+        var root = pkg.root;
+        if (root.data != .e_object) {
+            Output.errGeneric("package.json root must be an object", .{});
+            Global.exit(1);
+        }
+
+        var modified = false;
+        for (args) |arg| {
+            const eq_pos = strings.indexOf(arg, "=") orelse {
+                Output.errGeneric("Invalid argument: {s} (expected key=value)", .{arg});
+                Global.exit(1);
+            };
+
+            const key = arg[0..eq_pos];
+            const value = arg[eq_pos + 1 ..];
+
+            if (key.len == 0) {
+                Output.errGeneric("Empty key in argument: {s}", .{arg});
+                Global.exit(1);
+            }
+
+            if (value.len == 0) {
+                Output.errGeneric("Empty value in argument: {s}", .{arg});
+                Global.exit(1);
+            }
+
+            try setValue(ctx.allocator, &root, key, value, parse_json);
+            modified = true;
+        }
+
+        if (modified) {
+            try savePackageJson(ctx.allocator, path, root, &pkg);
+        }
+    }
+
+    fn execDelete(ctx: Command.Context, pm: *PackageManager, args: []const string, cwd: []const u8) !void {
+        _ = pm;
+        if (args.len == 0) {
+            Output.errGeneric("<blue>bun pm pkg <b>delete<r> expects key args", .{});
+            Global.exit(1);
+        }
+
+        const path = try findPackageJson(ctx.allocator, cwd);
+        defer ctx.allocator.free(path);
+
+        const pkg = try loadPackageJson(ctx, ctx.allocator, path);
+        defer ctx.allocator.free(pkg.contents);
```

Comment:
`const path = ` and `const pkg = ` should be pulled out of exec<subcommand> functions and passed in